internal/observability/events/event_audit_test.go

```diff
@@ -0,0 +1,278 @@
+package event
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/eventlogger"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_newAudit(t *testing.T) {
+	t.Parallel()
+	testNow := time.Now()
+
+	tests := []struct {
+		name         string
+		fromOp       Op
+		opts         []Option
+		want         *Audit
+		wantErrMatch *errors.Template
+	}{
+		{
+			name:         "missing-op",
+			wantErrMatch: errors.T(errors.InvalidParameter),
+		},
+		{
+			name:   "valid-no-opts",
+			fromOp: "valid-no-opts",
+			want: &Audit{
+				Version: AuditVersion,
+				Type:    string(ApiRequest),
```

Comment:
Should we type out `Audit.Type` so that this kind of type conversion is not necessary? Assuming this field will always be non-arbitrary.

Reply:
Audit.Type is string based on an effort to adhere to the Audit schema put forth on ENG-030 and CSL-121.  I used a defined type internally because it afford some advantages... but we should adhere to the previous work by the Nomad and Consul teams when it comes to the outbound audit event data.